test/merge_dune.ml

```diff
@@ -0,0 +1,171 @@
+      (* change name to b, reformat *)
+      write "dune" {|
+(executable
+ (name bbbb)
+ (libraries unix))
+|};
+      git_commit "second commit (fork)";
+
+      write "dune"
+        {|
+(executable
+ (name bbbb)
+ (libraries unix))
+
+(alias
+ (name runtest)
+ (action
+  (diff rebase.diff rebase.diff.gen)))
+|};
+
+      git_commit "third commit (fork)";
+      git_branch "old_branch1";
+      system "git rebase branch2 -q";
+      [%expect
+        {|
+        Auto-merging dune
+        CONFLICT (content): Merge conflict in dune
+        error: could not apply 2e00bc1... second commit (fork)
+        hint: Resolve all conflicts manually, mark them as resolved with
+        hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
+        hint: You can instead skip this commit: run "git rebase --skip".
+        hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
+        Could not apply 2e00bc1... second commit (fork)
+        Exit with 1 |}];
+      print_file "dune";
+      [%expect
+        {|
+        File dune
+
+        (executable
+        <<<<<<< HEAD
+        (name aaaa)
+        (libraries unix)
+        (public_name pppp))
+
+        (library
+        (name liblib))
+        =======
+         (name bbbb)
+         (libraries unix))
+        >>>>>>> 2e00bc1 (second commit (fork)) |}])
+
+let%expect_test "custom merge tool" =
+  within_temp_dir (fun () ->
+      git_init ();
+      system "%s setup-merge --merge-fmt-path %s --update" merge_fmt merge_fmt;
+      [%expect {||}];
+
+      write "dune" {|
+(executable
+(name aaaa)
+(libraries unix))
+|};
+      git_commit "first commit";
+      git_branch "branch1";
+
+      (* add public name *)
+      write "dune"
+        {|
+(executable
+(name aaaa)
+(libraries unix)
+(public_name pppp))
+|};
+      git_commit "second commit";
+
+      (* add library *)
+      write "dune"
+        {|
+(executable
+(name aaaa)
+(libraries unix)
+(public_name pppp))
+
+(library
+(name liblib))
+|};
+      git_commit "third commit";
+      git_branch "branch2";
+
+      git_checkout "branch1";
+      [%expect {| Switched to branch 'branch1' |}];
+      (* change name to b, reformat *)
+      write "dune" {|
+(executable
+ (name bbbb)
+ (libraries unix))
+|};
+      git_commit "second commit (fork)";
+
+      write "dune"
+        {|
+(executable
+ (name bbbb)
+ (libraries unix))
+
+(alias
+ (name runtest)
+ (action
+  (diff rebase.diff rebase.diff.gen)))
+|};
+      git_commit "third commit (fork)";
+      git_branch "old_branch1";
+
+      system "git rebase branch2 -q";
+      [%expect
+        {|
+        dropping a5a03b549f1159aac7629b550009dab46c7a42e0 third commit (fork) -- patch contents already upstream |}];
+      print_file "dune";
+      [%expect
+        {|
+        File dune
+        (executable
+         (name aaaa)
+         (libraries unix)
+         (public_name pppp))
+
+        (library
+         (name liblib)) |}])
```

Comment:
the output is not correct, it results into `branch2` formatted and the content of `old_branch1` is dropped. IIUC the final state should be

```
(executable
 (name bbbb)
 (libraries unix)
 (public_name pppp))

(library
(name liblib))

(alias
 (name runtest)
 (action
  (diff rebase.diff rebase.diff.gen)))
```